docs/dev/hacking-guide.md

```diff
@@ -0,0 +1,177 @@
+# Cluster cloud controller manage operator - Hacking guide
+
+## How to run a component locally for testing
+
+Prerequisites:
+
+```bash
+git checkout github.com/openshit/$repository_name
+cd $repository_name
+```
+
+Make sure your $KUBECONFIG is set properly, because it will be used to interact with your cluster.
+
+First step is to scale down the cluster version operator.
+
+“At the heart of OpenShift is an operator called the Cluster Version Operator. This operator watches the deployments and images related to the core OpenShift services, and will prevent a user from changing these details. If I want to replace the core OpenShift services I will need to scale this operator down.”
+
+```bash
+oc scale --replicas=0 deployment/cluster-version-operator -n openshift-cluster-version
+```
+
+Second step is scaling down the CCCMO. The operator watches for deployed resources to be running and will prevent them from scaling down.
+
+```bash
+oc scale deployment/cluster-cloud-controller-manager -n openshift-cloud-controller-manager-operator --replicas=0
+```
+
+Finally, once all has been scaled down you can compile and run the controller.
+
+You can copy or edit `hack/example-images.json` with the provider image you want to run.
+
+```bash
+make build
+ ./bin/cluster-cloud-controller-manager --images-json=hack/example-images.json
+```
+
+## How to build the operator in a container for remote testing
+
+Prerequisites:
+
+```bash
+git checkout github.com/openshit/$repository_name
+cd $repository_name
+```
+
+First you’ll need to build an image with your changes.
+
+```bash
+IMG="quay.io/your-repo/cluster-cloud-controller-manager-operator:latest" make images
+docker push quay.io/your-repo/cluster-cloud-controller-manager-operator:latest
+```
+
+Second step, similar to running the component locally we need to scale down CVO.
+
+```bash
+oc scale --replicas=0 deployment/cluster-version-operator -n openshift-cluster-version
+```
+
+Next, edit machine-api-operator-images ConfigMap and place a link to your image there.
+
+The clsuter-cloud-controller-manager-operator project contains all the resources and components that are used by the CCCMO. There is a ConfigMap named cloud-controller-manager-images. This ConfigMap contains references to all the images used by the CCCMO, it uses these to deploy the CCM and ensure they are the proper images. The ConfigMap looks something like this:
```

Comment:
Suggested typo and a bit more context for the last bit of this
```suggestion
The cluster-cloud-controller-manager-operator project contains all the resources and components that are used by the CCCMO.
There is a ConfigMap named cloud-controller-manager-images.
This ConfigMap contains references to all the images used by the CCCMO.
It uses these images to deploy the CCM to various platforms and ensure that the running images are from the correct release payload.

The ConfigMap looks something like this:
```

Reply:
similar to above, we should wrap exact names that would be seen in the cluster with block quotes. in this case the operator project name, and configmap